docs/schemas/trace-analytics/readme.md

```diff
@@ -0,0 +1,91 @@
+2. A writer built on schema 1.1 but writing to a cluster containing schema 1.2 data will succeed
+3. A writer built on schema 1.0 but writing to a cluster containing schema 2.0 data will succeed
+4. A writer built on schema 2.0 but writing to a cluster containing schema 1.0 data will succeed
+
+### Handling minor version updates
+
+Minor version updates will occur as new fields are needed to support new Trace Analytics features. The steps to update a schema minor version are to:
+
+1. Ensure both TA and DP owners are aligned with requirements
+2. Update the schema in the schema repo
+    1. Add new fields to the schema JSON
+    2. Increment the `version` field of the schema by 1
+    3. Update the documentation to describe the new fields
+3. Add test data to the TA plugin test suite
+    1. Don't update existing data in place, instead add new data following the new schema version. The test suite is expected to pass with a range of minor versions being tested at the same time.
+4. Update DP to start emitting documents following the new schema version
+5. Add new features to the TA plugin utilizing the new data
+
+### Handling Major version updates
+
+As schema owners, we will do our best to avoid introducing major version changes. However as our schemas are heavily tied to the OpenTelemetry spec, there is always the risk of an upstream backwards-incompatible change requiring a major version increase.
+
+Due to the potentially disjointed release schedules of both OpenSearch and the managed offering, we need to ensure that rolling out a major version change is carefully planned.
+
+A typical migration plan will first make Data Prepper artifacts available so that users can start ingesting their data to the new index. To prevent data loss during the migration period, users can be encouraged to simultaneously write to both the old and new indexes. This can be done by either running both old and new versions of Data Prepper side-by-side, or perhaps Data Prepper itself can be updated to write to dual indexes (TODO). Once users have the ability to write to the new index, Trace Analytics plugin updates will be made available which make use of the new major version index.
+
+The steps to handle a schema major version update are to:
+
+1. Update Data Prepper to use the new schema. Increment the Data Prepper major version and make new artifacts available to users.
+    * This must *always be done first*. Plugin changes cannot go out before Data Prepper artifacts are made available.
+    * Encourage users to start using the new Data Prepper version ahead of the plugin release. These will allow the user to upgrade their Trace Analytics plugin and immediately have new major version data to work with.
+2. Update the Trace Analytics plugin to read from the new indexes. Increment the plugin version and release to OpenSearch and/or the managed offering.
+    * Communicate to users the need to use the new version of Data Prepper after upgrading their TA plugin
```

Comment:
Do you mean user should never return to old version of data prepper after upgrading TA? I would make it more explicit in description.

Reply:
TBD - I'd expect such instructions or requirements to be in the release notes or announcement blog posts, not here. I might even cut the migration strategies from this doc as they're not strictly related to schema versioning 🤔

Reply:
Sure. Makes sense